proxy/http/HttpSM.cc

```diff
@@ -5477,20 +5477,21 @@ HttpSM::handle_server_setup_error(int event, void *data)
     break;
 
   case VC_EVENT_INACTIVITY_TIMEOUT:
-    // If we're writing the request and get an inactivity timeout
-    //   before any bytes are written, the connection to the
-    //   server failed
-    // In case of TIMEOUT, the iocore sends back
-    // server_entry->read_vio instead of the write_vio
-    // if (vio->op == VIO::WRITE && vio->ndone == 0) {
-    if (server_entry->write_vio && server_entry->write_vio->nbytes > 0 && server_entry->write_vio->ndone == 0) {
-      t_state.current.state = HttpTransact::CONNECTION_ERROR;
-    } else {
-      t_state.current.state = HttpTransact::INACTIVE_TIMEOUT;
-    }
-    // Clean up the vc_table entry so any events in play to the timed out server vio
-    // don't get handled.  The connection isn't there.
     if (server_entry) {
+      // If we're writing the request and get an inactivity timeout
+      //   before any bytes are written, the connection to the
+      //   server failed
+      // In case of TIMEOUT, the iocore sends back
+      // server_entry->read_vio instead of the write_vio
+      // if (vio->op == VIO::WRITE && vio->ndone == 0) {
+      if (server_entry->write_vio && server_entry->write_vio->nbytes > 0 && server_entry->write_vio->ndone == 0) {
+        t_state.current.state = HttpTransact::CONNECTION_ERROR;
+      } else {
+        t_state.current.state = HttpTransact::INACTIVE_TIMEOUT;
```

Comment:
We should set `t_state.current.state = HttpTransact::INACTIVE_TIMEOUT` if even if server_entry is nullptr

Reply:
In that case, is that really inactive timeout? It might be a connection error.